audit/log_test.go

```diff
@@ -0,0 +1,194 @@
+package audit
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/go-ns/handlers/requestID"
+	"github.com/ONSdigital/go-ns/log"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type eventResults struct {
+	name string
+	cKey string
+	data log.Data
+}
+
+func TestErrorLog(t *testing.T) {
+
+	var eventRes eventResults
+	log.Event = func(name string, correlationKey string, data log.Data) {
+		eventRes = eventResults{
+			name: name,
+			cKey: correlationKey,
+			data: data,
+		}
+	}
+
+	ctx := context.Background()
+
+	errTest := errors.New("test error")
+
+	Convey("with empty context", t, func() {
+		LogError(ctx, errTest, nil)
+
+		So(eventRes, ShouldResemble, eventResults{
+			name: "error",
+			cKey: "",
+			data: log.Data{"message": "test error", "error": errTest},
+		})
+
+		eventRes = eventResults{}
+	})
+
+	Convey("context with request id", t, func() {
+		contextWithRequestID := context.WithValue(ctx, requestID.ContextKey, "request id")
+		LogError(contextWithRequestID, errTest, nil)
+
+		So(eventRes, ShouldResemble, eventResults{
+			name: "error",
+			cKey: "request id",
+			data: log.Data{"message": "test error", "error": errTest},
+		})
+
+		eventRes = eventResults{}
+	})
+
+	Convey("context with user", t, func() {
+		contextWithUser := context.WithValue(ctx, common.UserIdentityKey, "user email")
+		LogError(contextWithUser, errTest, nil)
+
+		So(eventRes, ShouldResemble, eventResults{
+			name: "error",
+			cKey: "",
+			data: log.Data{"message": "test error", "error": errTest, reqUser: "user email"},
+		})
+
+		eventRes = eventResults{}
+	})
+
+	Convey("context with caller id", t, func() {
+		contextWithCaller := context.WithValue(ctx, common.CallerIdentityKey, "api service")
+		LogError(contextWithCaller, errTest, nil)
+
+		So(eventRes, ShouldResemble, eventResults{
+			name: "error",
+			cKey: "",
+			data: log.Data{"message": "test error", "error": errTest, reqCaller: "api service"},
+		})
+
+		eventRes = eventResults{}
+	})
+
+	Convey("context with request, user and caller ids", t, func() {
+		ctx := getContextWithCallerUserAndRequestIDcontext()
+		LogError(ctx, errTest, nil)
+
+		So(eventRes, ShouldResemble, eventResults{
+			name: "error",
+			cKey: "request id",
+			data: log.Data{
+				"message": "test error",
+				"error":   errTest,
+				reqCaller: "api service",
+				reqUser:   "user email",
+			},
+		})
+
+		eventRes = eventResults{}
+	})
+}
+
+func TestInfoLog(t *testing.T) {
+
+	var eventRes eventResults
+	log.Event = func(name string, correlationKey string, data log.Data) {
+		eventRes = eventResults{
+			name: name,
+			cKey: correlationKey,
+			data: data,
+		}
+	}
```

Comment:
this is all repeated from the previous test, could they be moved to package-level?